tests/test_utils.py

```diff
@@ -94,3 +100,25 @@ def test_make_indices_patch(client, cache, name, content, is_patch, expected_ind
         assert expected_indices.line_index_end == match_indices.line_index_end
         assert expected_indices.index_start == match_indices.index_start
         assert expected_indices.index_end == match_indices.index_end
+
+
+def test_retrieve_client_invalid_api_url():
+    """
+    GIVEN a GITGUARDIAN_API_URL missing its https scheme
+    WHEN retrieve_client() is called
+    THEN it raises a ClickException
+    """
+    with pytest.raises(click.ClickException):
+        with mock.patch.dict(os.environ, {"GITGUARDIAN_API_URL": "no-scheme.com"}):
+            retrieve_client(Config())
+
+
+def test_retrieve_client_invalid_api_key():
+    """
+    GIVEN a GITGUARDIAN_API_KEY with a non-latin-1 character
+    WHEN retrieve_client() is called
+    THEN it raises a ClickException
+    """
+    with pytest.raises(click.ClickException):
```

Comment:
Can you assert the exception message content to emphasize what will be the error message displayed to the user please :pray: ?